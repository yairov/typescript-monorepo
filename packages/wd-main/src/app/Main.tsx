import React from 'react';
import Flexible from 'wd-flexible';
import calculate from 'wd-common/utils/calculate';
import CommonComponent from 'wd-common/components/CommonComponent';

export default function Main() {
  return (
    <div>
      <h1>Hey! I'm the "Main" project</h1>
      <p>
        I'm using the "Flexible" project which can be ran separately.
        <br/><br/>
        We both use the "Common" project.
        <br/><br/>
        Here is the "Flexible" project":
      </p>
      <CommonComponent text="CommonComponentInMain"/>
      <div>
        Calculate using number: {calculate(1)}
      </div>
      <div style={{border: '1px solid black', padding: 20}}>
        <Flexible/>
      </div>
    </div>
  );
}
