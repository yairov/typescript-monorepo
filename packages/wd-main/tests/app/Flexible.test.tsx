import React from 'react';
import {render} from '@testing-library/react';

import Main from 'wd-main/src/app/Main';

describe('Main', () => {
  test('simple', () => {
    const {asFragment} = render(<Main/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
