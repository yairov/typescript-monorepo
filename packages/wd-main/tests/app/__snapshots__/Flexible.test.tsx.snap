// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main simple 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hey! I'm the "Main" project
    </h1>
    <p>
      I'm using the "Flexible" project which can be ran separately.
      <br />
      <br />
      We both use the "Common" project.
      <br />
      <br />
      Here is the "Flexible" project":
    </p>
    <div>
      hey! I'm a div that's being used in both projects! CommonComponentInMain
    </div>
    <div>
      Calculate using number: 2
    </div>
    <div
      style="border: 1px solid black; padding: 20px;"
    >
      <div>
        <h1>
          Hey! I'm the "Flexible" project
        </h1>
        <p>
          I can be a dependency of "Main" project or ran independently.
          <br />
          <br />
          We both use the "Common" project.
        </p>
        <div>
          hey! I'm a div that's being used in both projects! CommonComponentInFlexible
        </div>
        <div>
          Calculate using string: str1
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
